### PR TITLE
Python CDK: Delete deprecated authenticators

### PR DESCRIPTION
**Pull request by girarda**
## What
Delete `sources.streams.http.auth` modules from from the Python CDK. Connectors should instead use the authenticators from the [requests_native_auth](https://github.com/airbytehq/airbyte/tree/master/airbyte-cdk/python/airbyte_cdk/sources/streams/http/requests_native_auth) package

The PR is targeting alex/cdk_1.0.0 as a dev branch so we can batch multiple changes in a single breaking change.

## Can this PR be safely reverted and rolled back?
- [X] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Review by maxi297 — APPROVED** [review]
LGTM!

File: airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py [hunk 1]
```diff
@@ -40,17 +39,13 @@ class HttpStream(Stream, ABC):
     page_size: Optional[int] = None  # Use this variable to define page size for API http requests with pagination support
 
     # TODO: remove legacy HttpAuthenticator authenticator references
```
Comment (erohmensing) [inline airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py:41]:
```suggestion
```
This is probably the only one but might be worth a quick comment search for references!

File: airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py [hunk 2]
```diff
@@ -134,10 +129,6 @@ def retry_factor(self) -> float:
         """
         return 5
 
-    @property
-    def authenticator(self) -> HttpAuthenticator:
-        return self._authenticator
-
```
Comment (erohmensing) [inline airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py:140]: What's the upgrade path if you were using this property? I see a decent amount of `self.authenticator` usage in the connectors

Comment (erohmensing) [reply airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py:140]: I've sort of answered my question: This is only set if you're using one of the deprecated classes that aren't coming from `AuthBase`. 

It looks l like what the migrated connectors are doing is accessing `_session.auth` directly, which the connectors could do to migrate over (which is still a bit weird as `_session` is a "private" property

**Review by erohmensing — APPROVED** [review]
Peace out, version 1.20 code

File: airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py (hunk 2 above)
Comment (girarda) [reply airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py:140]: yeah using the `_session.auth` param is a little off. I'm not overly concerned about it as the new HttpClient is reusable so there shouldn't be a need for copying an authenticator from a stream to another. Next step is to use the client in the declarative module then we'll update the `HttpStream`

Comment (erohmensing) [reply airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py:140]: Sweet, sounds like a good plan!

**Pull request merged**